web-api/migration-terraform/main/lambdas/migration.js

```diff
@@ -1,41 +1,55 @@
 const AWS = require('aws-sdk');
 const { get } = require('lodash');
+const { migrateRecords: migrations } = require('./migration-segments');
 
 const dynamodb = new AWS.DynamoDB({
   maxRetries: 10,
   retryDelayOptions: { base: 300 },
 });
 
-const documentClient = new AWS.DynamoDB.DocumentClient({
+const docClient = new AWS.DynamoDB.DocumentClient({
   endpoint: 'dynamodb.us-east-1.amazonaws.com',
   region: 'us-east-1',
   service: dynamodb,
 });
 
-const processItems = async items => {
+const processItems = async ({ documentClient, items, migrateRecords }) => {
   const promises = [];
+
+  items = await migrateRecords({ documentClient, items });
+
   for (const item of items) {
-    // TODO: your migration code would go here
-    const result = documentClient
-      .put({
-        Item: item,
-        TableName: process.env.DESTINATION_TABLE,
-      })
-      .promise();
-    promises.push(result);
+    promises.push(
+      documentClient
+        .put({
+          Item: item,
+          TableName: process.env.DESTINATION_TABLE,
+        })
+        .promise(),
+    );
   }
   await Promise.all(promises);
 };
 
-exports.handler = async event => {
+const getFilteredGlobalEvents = event => {
   const { Records } = event;
-  const items = Records.filter(record => {
+  return Records.filter(record => {
     // to prevent global tables writing extra data
     const NEW_TIME_KEY = 'dynamodb.NewImage.aws:rep:updatetime.N';
     const OLD_TIME_KEY = 'dynamodb.OldImage.aws:rep:updatetime.N';
     const newTime = get(record, NEW_TIME_KEY);
     const oldTime = get(record, OLD_TIME_KEY);
     return newTime && newTime !== oldTime;
   }).map(item => AWS.DynamoDB.Converter.unmarshall(item.dynamodb.NewImage));
-  await processItems(items);
+};
```

Comment:
Would these writes need to happen _after_ the migration is complete? 

In other words, is there risk in this scenario happening?

- a b/g migration is kicked off
- a user makes an update to a Case DDB record
- this lambda picks it up and writes it to the destination table _before_ the migration-segments Lambda does
- the migration-segments.js Lambda overwrites the newer record with stale information.

Reply:
^ Fantastic question. I would love to learn more about this if y'all have a zoom conversation about it!

Reply:
I addressed this issue by adding a conditional write on the segments-migration script.  If the record already exists in the destination table (since we always plan to keep the destination table empty before a migration), it will skip migrating the record since a user more than likely did a manual update on a record